libs/blocks/global-navigation/utilities/getUserEventHistory.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable no-promise-executor-return, no-async-promise-executor */
+import { getConfig } from '../../../utils/utils.js';
+
+let userEventHistory;
+const getUserEventHistory = async (id) => {
+  if (userEventHistory) return userEventHistory;
+  if (!window.adobeIMS?.isSignedInUser()) return Promise.reject(new Error('User not signed in'));
+  let resolve;
+  let reject;
+  userEventHistory = userEventHistory || new Promise((_resolve, _reject) => {
+    resolve = _resolve;
+    reject = _reject;
+  });
+  const profile = await window.adobeIMS.getProfile();
+  const apiUrl = getConfig().env.name === 'prod'
+    ? 'https://www.adobe.com/gating/api/realtime-profile'
+    : 'https://www.stage.adobe.com/gating/api/realtime-profile';
+  const res = await window
```

Comment:
can't we just call fetch() ?

Reply:
Yup, we can change that ... @overmyheadandbody do you want to batch that with other smaller improvements for this PR?